rtt/base/BufferBase.cpp

```diff
@@ -38,7 +38,43 @@
 
 
 #include "BufferBase.hpp"
+#include "../ConnPolicy.hpp"
 
+using namespace RTT;
 using namespace RTT::base;
 
 BufferBase::~BufferBase() {}
+
+BufferBase::Options::Options()
+    : circular_(false)
+    , max_threads_(2)
+    , multiple_writers_(false)
+    , multiple_readers_(false)
+{}
+
+BufferBase::Options::Options(bool circular)
+    : circular_(circular)
+    , max_threads_(2)
+    , multiple_writers_(false)
+    , multiple_readers_(false)
+{}
+
+BufferBase::Options::Options(const ConnPolicy &policy)
+    : circular_(policy.type == ConnPolicy::CIRCULAR_BUFFER)
+    , max_threads_(2)
+//    , multiple_writers_(policy.read_policy == ReadShared)
```

Comment:
Yes: This PR must be considered as a part of the bigger data flow refactoring in #114. The complete patch sets the `multiple_writers_` and `multiple_readers_` flags from the `read_policy`/`write_policy` fields of the connection policy. I only commented the initializers here because the new fields are not yet available without #118.